### ISSUE TITLE
How come `onreadystatechange` is the only event handler defined by default ?

### ISSUE THREAD

**Issue by shacharz** [whatwg/xhr#222]
https://xhr.spec.whatwg.org/#idl-index

`onreadystatechange` is defined in the constructor but not other event handlers.
Which causes it to have a propertyDescriptor while other event handlers don't.
How come ?

**Comment by annevk** [comment]
They are inherited from `XMLHttpRequestEventTarget`.

**Comment by shacharz** [comment]
> They are inherited from XMLHttpRequestEventTarget.

Yea, why was `XMLHttpRequestEventTarget` specified like that ?

**Comment by annevk** [comment]
Because there's also `XMLHttpRequestUpload` and we typically inherit from `EventTarget` where possible.

**Comment by shacharz** [comment]
Not sure I understand, I'm asking how come `onreadystatechange` is defined but `onload`,`onloadend`,`onprogress`, etc' aren't

**Comment by annevk** [comment]
They are defined, just higher up in the prototype chain.

**Comment by annevk** [comment]
Because they're higher up on the prototype chain, I can recommend https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain or asking on Stackoverflow.

(https://heycam.github.io/webidl/ defines the translation from IDL to JavaScript objects, if you want to know the details, but it might be hard to follow.)

**Comment by shacharz** [comment]
Will do, Thanks!
